mongodb-plugins/src/main/java/co/cask/hydrator/plugin/batch/sink/MongoDBBatchSink.java

```diff
@@ -76,6 +77,7 @@ public void transform(StructuredRecord input, Emitter<KeyValue<NullWritable, BSO
     MongoDBOutputFormatProvider(MongoDBSinkConfig config) {
       this.conf = new HashMap<>();
       conf.put("mongo.output.uri", config.connectionString);
+      conf.put("mapreduce.task.tmp.dir", "/tmp/" + UUID.randomUUID().toString());
```

Comment:
oh i see, because this is different for each sink... but this will break in setups where mapred-site.xml has been modified to something custom. You will then override it to /tmp/, which may not be available or writable.